acceptance/tests/lookup/merge_strategies.rb

```diff
@@ -0,0 +1,194 @@
+test_name 'C99903: merge strategies' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+  tmp_environment2 = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath2  = "#{environmentpath}/#{tmp_environment2}"
+
+  master_confdir = master.puppet('master')['confdir']
+  hiera_conf_backup = master.tmpfile(app_type)
+
+  teardown do
+    step "restore default global hiera.yaml" do
+      on(master, "mv #{hiera_conf_backup} #{master_confdir}/hiera.yaml")
```

Comment:
This teardown step should ensure that the `hiera_conf_backup` file is present beforehand. If the backup step below should fail to create the backup file, then the teardown will erroneously fail.